docs/workflows/common-workflow-steps/change-java-version.md

```diff
@@ -0,0 +1,105 @@
+
+- [Android Build Stacks - Java Version](/infrastructure/android-build-infrastructure#java-version)
+
+Note that available Java versions may vary based on the runner you are using.
+
+For self-hosted users, the versions listed for the cloud Appcircle runner may not apply. Refer to the FAQ below for guidance:
+
+- [How do I check available Java versions in the Appcircle runner?](#how-do-i-check-available-java-versions-in-the-appcircle-runner)
+
+:::tip Check Java Version for Android
+
+If you are unsure which Java version is required for your Android project, you can find it by following the steps in this FAQ:
+
+- [How can I check the Java version of my project?](#how-can-i-check-the-java-version-of-my-project)
+
+:::
+
+
+## Prerequisites
+
+There is no mandatory step for using this component.  However, if you have a step that necessitates changing the Java version (particularly build steps), the **Change Java Version** step should run before that step in the [workflow](/workflows).
+
+For reference, you can see the workflow sequence in the image below:
+
+<Screenshot url='https://cdn.appcircle.io/docs/assets/common-workflow-components-change-java-version-1.0.png' />
+
+:::caution
+
+If your runner is self-hosted, ensure that the selected Java version is available in your environment. If not, you must install the required version before the **Change Java Version** step.
+
+:::
+
+
+## Input Variables
+
+A required parameter for this step is outlined in the table below, along with its description.
+
+<Screenshot url='https://cdn.appcircle.io/docs/assets/common-workflow-components-change-java-version-2.0.png' />
+
+
+| Variable Name               | Description                                                                                                 | Status   |
+| --------------------------- | ----------------------------------------------------------------------------------------------------------- | -------- |
+| `$AC_SELECTED_JAVA_VERSION` | Specify the Java version to switch to. The default value is `17`. Options: `8`, `11`, `17`, `21`, and `23`. | Required |
+
+:::info
+
+If the selected Java version is not available on the runner, an error message will indicate that `this version is not found` and will display the available Java versions for your runner. Check available Java versions for cloud runner [here](/infrastructure/android-build-infrastructure#java-version).
+
+:::
+
+## Output Variables
+
+This step does not produce any visible output. However, if run successfully, it will update both the default Java version (`$JAVA_HOME`) and the system JDK version to the selected version.
+
+---
+
+To access the source code of this component, please use the following link:
+
+https://github.com/appcircleio/appcircle-change-java-version-component
+
```

Comment:
Please add the "---" here, It's a good way to separate the FAQ section from the main document and repo information.

Reply:
Sorry I forgot to send the last commit I made. That commit includes this change, so I'm closing it.